skt/runner.py

```diff
@@ -28,6 +28,70 @@
 from skt.misc import WaivingWrap
 
 
+class ConditionCheck:
+    def __init__(self, retval, **kwargs):
+        self.retval = retval
+        self.kwargs = kwargs
+
+    def __call__(self, task, waiving, is_task_waived_func, prev_task):
+        """ Evaluates the condition and return retval if matched, else None.
+
+            Args:
+                task: defusedxml of the task node
+                waiving: bool, determines if waiving is enabled
+                is_task_waived_func: function used to test whether the task
+                                     is waived
+                prev_task: task that was run before this one, or None if this
+                           task is the first task in the recipe
+        """
+        task_results = {
+            'result': task.attrib.get('result'),
+            'status': task.attrib.get('status'),
+            'waived': waiving and is_task_waived_func(task),
+            'prev_panicked_and_waived':
+                prev_task is not None and (
+                waiving and is_task_waived_func(prev_task) and
+                prev_task.attrib.get('result') == 'Panic')
+        }
+
+        if not self.kwargs:
+            # don't match empty conditions as satisfied
+            return None
+
+        for arg in self.kwargs:
+            if task_results[arg] != self.kwargs[arg]:
+                # the status entry doesn't match all the conditions
+                return None
+
+        # the status entry matches all the conditions
+        return self.retval
+
+
+result_condition_checks = [
+    # This contains objects that will return <retval> (first parameter), when
+    # all the specified conditions are met. Empty conditions with no keywords
+    # are never met.
+
+    # Previous task was waived and panicked, which causes the next
+    # task to abort. The task is waived for a reason, return
+    # SKT_SUCCESS.
+    ConditionCheck(SKT_SUCCESS, result='Fail', waived=False, status='Aborted',
```

Comment:
Shouldn't this be `result=Warn` ?

Reply:
I think so. The original code would actually allow warn/fail/panic in this check...